### PR TITLE
Simplify access token matching and add docs URL

### PR DESCRIPTION
**Pull request by willdollman**
Simplify token matching format, as the previous length-based validation missed an edge case. Also remove `sgph_` prefix as we're going ahead with `sgp_`.

~URL in comment is pending https://github.com/sourcegraph/sourcegraph/pull/57722 merging.~

## Test plan

- CI

### CODE REVIEW

File: src/main/java/com/sourcegraph/common/AuthorizationUtil.java [hunk 1]
```diff
@@ -4,11 +4,7 @@
 
 public class AuthorizationUtil {
   public static boolean isValidAccessToken(@NotNull String accessToken) {
-    return accessToken.isEmpty()
-        || accessToken.length() == 40
-        || (accessToken.startsWith("sgp_") && accessToken.length() == 44)
-        // See https://docs.google.com/document/d/1aC4gHB8Q5lurwVhc8SCxznR0blNJMKo7yIkpP7WShno
-        || (accessToken.startsWith("sgp_") && accessToken.length() == 61)
-        || (accessToken.startsWith("sgph_") && accessToken.length() == 62);
+    // Sourcegraph access token formats: https://docs.sourcegraph.com/dev/security/secret_formats
```
Comment (mkondratek) [inline src/main/java/com/sourcegraph/common/AuthorizationUtil.java:7]: image

Should the page be yet available? 🤔

Comment (willdollman) [reply src/main/java/com/sourcegraph/common/AuthorizationUtil.java:7]: Not quite yet I'm afraid!

> URL in comment is pending https://github.com/sourcegraph/sourcegraph/pull/57722 merging.

Comment (willdollman) [reply src/main/java/com/sourcegraph/common/AuthorizationUtil.java:7]: PR has merged and link now works @mkondratek https://docs.sourcegraph.com/dev/security/secret_formats

Comment (mkondratek) [reply src/main/java/com/sourcegraph/common/AuthorizationUtil.java:7]: Thanks! It looks good. 

I am curious... what about `sgd_`, `slk_`, `sgs_` tokens? They are mentioned in the docs. However, they are invalid in the sense of this method. Is it ok? 🤔

Comment (willdollman) [reply src/main/java/com/sourcegraph/common/AuthorizationUtil.java:7]: Yep not a problem. The jetbrains extension only needs to use access tokens (`sgp_`) which are used to talk to a sourcegraph instance. These other tokens are used for various other backend authentication purposes so don't need to be supported here 😄

**Review by mkondratek — APPROVED** [review]
ok, thanks for the explanations. makes sense. lgtm
